### PR TITLE
Remove Request Preview Feature Flag

### LINKED ISSUE #898: Remove Request Preview Feature Flag & Update Tests
In `request-v2` branch

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #898

## Description
Removes all mentions of the RequestPreview feature flag, and changes testing functionality to test functionality that would previously only work with the feature flag on.

## Proposed Changes

-

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/tests/acceptance/report-test.js [hunk 1]
```diff
@@ -147,21 +141,15 @@ module('Acceptance | Navi Report', function(hooks) {
...
-    await dayTimeGrain.reset();
+    assert.dom('.filter-builder__subject').hasText('Date Time (Day)');
 
     // uncheck the day timegrain
-    await clickItem('timeGrain', 'Day');
-
-    dayTimeGrain = await getTimeGrainCheckbox('Day');
-    assert.dom(dayTimeGrain.item).isNotChecked('Day timegrain is unchecked after clicking on it');
-    await dayTimeGrain.reset();
+    await click('.navi-column-config-item__remove-icon[aria-label="delete time-dimension Date Time (Day)"]');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-test.js:152]: Can we update the comment to remove `date time` column

File: packages/reports/tests/acceptance/report-test.js [hunk 2]
```diff
@@ -175,9 +163,7 @@ module('Acceptance | Navi Report', function(hooks) {
       .dom('.navi-report__revert-btn')
       .isNotVisible('After navigating away and back to the route, the Revert button disappears');
 
-    dayTimeGrain = await getTimeGrainCheckbox('Day');
-    assert.dom(dayTimeGrain.item).isChecked('After navigating away and back to the route, changes are reverted');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-test.js:179]: Can we add an assert that the `date time` column is present

File: packages/reports/tests/acceptance/report-test.js [hunk 3]
```diff
@@ -191,7 +177,8 @@ module('Acceptance | Navi Report', function(hooks) {
     );
 
     // Remove a metric
-    await clickItem('timeGrain', 'Week');
+    await click('.navi-column-config-item__trigger');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-test.js:180]: is that comment correct?

File: packages/reports/tests/acceptance/report-test.js [hunk 4]
```diff
@@ -1479,92 +1470,8 @@ module('Acceptance | Navi Report', function(hooks) {
...
-    await clickItem('dimension', 'Operating System');
-
-    assert.deepEqual(
-      await getAllSelected('dimension'),
-      ['Day', 'Operating System', 'Property'],
-      'Adding a dimension changes selected dimensions'
-    );
-
-    // Add selected dimension as filter
-    await clickItemFilter('dimension', 'Operating System');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-test.js:1510]: Are we testing adding a dim filter in the test below?

File: packages/reports/tests/acceptance/report-test.js [hunk 5]
```diff
@@ -1611,89 +1518,11 @@ module('Acceptance | Navi Report', function(hooks) {
...
-    await clickItem('metric', 'Total Clicks');
-
-    assert.deepEqual(
-      await getAllSelected('metric'),
-      ['Ad Clicks', 'Nav Link Clicks', 'Total Clicks'],
-      'Adding a metric changes selected metrics'
-    );
-
-    // Add selected metric as filter
-    await clickItemFilter('metric', 'Total Clicks');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-test.js:1647]: same here

File: packages/reports/tests/acceptance/report-visualizations-test.js [hunk 6]
```diff
@@ -267,7 +266,8 @@ module('Acceptance | navi-report - report visualizations', function(hooks) {
       );
 
     //remove the metric
-    await clickItem('metric', 'Platform Revenue');
+    await click('.navi-column-config-item__remove-icon[aria-label="delete metric Platform Revenue (EUR)"]');
```
Comment (jkusa) [inline packages/reports/tests/acceptance/report-visualizations-test.js:269]: Can we update the comment above?

File: packages/reports/tests/integration/components/metric-selector-test.js [hunk 7]
```diff
@@ -100,132 +90,23 @@ module('Integration | Component | metric selector', function(hooks) {
...
-    config.navi.FEATURES.enableRequestPreview = true;
-
     await render(TEMPLATE);
 
-    assert
-      .dom('.navi-list-selector__show-link')
-      .doesNotExist('Show Selected toggle is hidden if enableRequestPreview flag is turned on');
-
-    config.navi.FEATURES.enableRequestPreview = originalFeatureFlag;
+    assert.dom('.navi-list-selector__show-link').doesNotExist('Show Selected toggle is hidden');
```
Comment (jkusa) [inline packages/reports/tests/integration/components/metric-selector-test.js:97]: Can we remove this test

File: packages/reports/tests/integration/components/navi-collection-test.js [hunk 8]
```diff
@@ -93,10 +93,6 @@ module('Integration | Component | navi collection', function(hooks) {
     //Reset to all filter
     await selectChoose('.navi-collection__filter-trigger', 'All');
 
-    assert
```
Comment (jkusa) [inline packages/reports/tests/integration/components/navi-collection-test.js:96]: Why was this removed?

File: packages/reports/tests/integration/components/navi-list-selector-test.js [hunk 9]
```diff
@@ -65,85 +64,19 @@ module('Integration | Component | navi list selector', function(hooks) {
...
-    config.navi.FEATURES.enableRequestPreview = true;
-
     await render(TEMPLATE);
 
-    assert
-      .dom('.navi-list-selector__show-link')
-      .doesNotExist('Show Selected toggle is hidden if enableRequestPreview flag is turned on');
-
-    config.navi.FEATURES.enableRequestPreview = originalFeatureFlag;
+    assert.dom('.navi-list-selector__show-link').doesNotExist('Show Selected toggle is hidden');
```
Comment (jkusa) [inline packages/reports/tests/integration/components/navi-list-selector-test.js:75]: Can this test be removed?

File: packages/reports/addon/templates/components/metric-selector.hbs [hunk 10]
```diff
@@ -38,15 +38,6 @@
         </NaviIcon>
 
         <div class="grouped-list__icon-set {{if (not (can-having metric)) "grouped-list__icon-set--no-filter"}}">
-          {{#if (and metric.hasParameters (not (feature-flag "enableRequestPreview")))}}
-            <MetricConfig
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/templates/components/metric-selector.hbs:42]: We should be able to now

File: packages/core/tests/unit/models/bard-request/request-test.js [hunk 11]
```diff
@@ -550,7 +540,7 @@ module('Unit | Model Fragment | BardRequest - Request', function(hooks) {
 
     request.addRequestMetricWithParam(newMetric, { currency: 'USD' });
 
-    assert.equal(request.metrics.length, 3, 'The final metric is not added since it already exists in the request');
+    assert.equal(request.metrics.length, 4, 'The final metric is added because of requests preview');
```
[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/core/tests/unit/models/bard-request/request-test.js:543]: This file will be deleted anyways so we can probably ignore it

File: packages/reports/addon/components/dimension-selector.js [hunk 12]
```diff
@@ -208,18 +203,13 @@ export default class DimensionSelector extends Component {
...
-        return this.onAddTimeGrain?.(this.selectedTimeGrain || this.defaultTimeGrain);
-      } else if (type === 'Dimension') {
-        actionHandler = 'Add';
-      }
-    } else {
-      actionHandler = this.itemsChecked[item.id] ? 'Remove' : 'Add';
+    if (type === 'TimeGrain') {
+      return this.onAddTimeGrain?.(this.selectedTimeGrain || this.defaultTimeGrain);
+    } else if (type === 'Dimension') {
+      actionHandler = 'Add';
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/dimension-selector.js:212]: Likely not

File: packages/reports/tests/acceptance/report-test.js (hunk 3 above)
Comment (holderb5) [reply packages/reports/tests/acceptance/report-test.js:180]: Yes, the select choose on line 181 removes the metric

File: packages/reports/tests/acceptance/report-test.js (hunk 4 above)
Comment (holderb5) [reply packages/reports/tests/acceptance/report-test.js:1510]: That seems to be the goal, would you like me to add those into the test that used to be enabled request preview?

File: packages/reports/addon/templates/components/metric-selector.hbs (hunk 10 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/templates/components/metric-selector.hbs:42]: I'll do that

File: packages/reports/tests/acceptance/report-visualizations-test.js [hunk 13]
```diff
@@ -265,7 +265,7 @@ module('Acceptance | navi-report - report visualizations', function(hooks) {
         'Metric with first parameter is sorted desc after sorting by both parameters'
       );
 
-    //remove the metric
+    //remove both the ERU metric and the USD metric
```
Comment (kevinhinterlong) [inline packages/reports/tests/acceptance/report-visualizations-test.js:268]:
```suggestion
    //remove both the EUR metric and the USD metric
```

File: packages/reports/addon/components/dimension-selector.js [hunk 14]
```diff
@@ -204,17 +200,11 @@ export default class DimensionSelector extends Component {
...
     if (type === 'TimeGrain') {
       return this.onAddTimeGrain?.(this.selectedTimeGrain || this.defaultTimeGrain);
-    } else if (type === 'Dimension') {
-      actionHandler = 'Add';
     }
 
-    if (actionHandler) {
-      this[`on${actionHandler}${type}`]?.(item);
-    }
+    this[`onAdd${type}`]?.(item);
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/dimension-selector.js:207]:
```suggestion
    this.onAddDimension(item);
```

File: packages/reports/addon/components/dimension-selector.js [hunk 15]
```diff
@@ -204,17 +200,11 @@ export default class DimensionSelector extends Component {
     target && target.focus(); // firefox does not focus a button on click in MacOS specifically
     const type = item.dateTimeDimension ? 'TimeGrain' : 'Dimension';
 
-    let actionHandler;
-
     if (type === 'TimeGrain') {
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/dimension-selector.js:203]: We could change this check here to just be `item.dateTimeDimension `

File: packages/reports/addon/components/report-view.js [hunk 16]
```diff
@@ -42,8 +42,6 @@ class ReportView extends Component {
   get classNames() {
     const classNames = ['report-view'];
 
-    classNames.push('report-view--request-preview');
-
     return classNames;
   }
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/report-view.js]:
```js
classNames = ['report-view']
```

File: packages/reports/addon/templates/components/metric-selector.hbs [hunk 17]
```diff
@@ -6,7 +6,7 @@
     @selected={{this.selectedMetrics}}
     @title="Metrics"
     @contentClass="navi-list-selector__content--metric"
-    class="navi-list-selector--request-preview"
+    class="navi-list-selector"
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/metric-selector.hbs:9]:
```suggestion
```
We should be able to get rid of this now since `<NaviListSelector />` should already apply the class

File: packages/reports/addon/templates/components/dimension-selector.hbs [hunk 18]
```diff
@@ -5,7 +5,7 @@
   @selected={{this.selectedColumnsAndFilters}}
   @title="Dimensions"
   @contentClass="navi-list-selector__content--dimension"
-  class="navi-list-selector--request-preview"
+  class="navi-list-selector"
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/dimension-selector.hbs:8]: This one too

File: packages/reports/tests/acceptance/report-test.js (hunk 3 above)
Comment (kevinhinterlong) [reply packages/reports/tests/acceptance/report-test.js:179]: ?

Comment (holderb5) [reply packages/reports/tests/acceptance/report-test.js:179]: wait no that would add the Week dimension again, sorry let me fix that.

File: packages/reports/addon/components/dimension-selector.js (hunk 15 above)
Comment (holderb5) [reply packages/reports/addon/components/dimension-selector.js:203]: I dont think thats the case, I changed it as a test, and then all of the timegrain tests start failing

Comment (holderb5) [reply packages/reports/addon/components/dimension-selector.js:203]: Or do you mean just making the conditional if(item.dateTimeDimension) {

Comment (kevinhinterlong) [reply packages/reports/addon/components/dimension-selector.js:203]: Yea just update the conditional logic to take out the extra step

**Comment by kevinhinterlong** [comment]
Closing in favor of #967 request-v2 specific branch
